membership_prorate_variable_period/models/product_template.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2016 Tecnativa - Antonio Espinosa
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+import math
+import datetime
+import calendar
+from dateutil.relativedelta import relativedelta
+from odoo import _, api, exceptions, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.multi
+    def _get_next_date(self, date, qty=1):
+        res = super(ProductTemplate, self)._get_next_date(date)
+        if self.membership_interval_unit == 'days':
+            raise exceptions.Warning(
+                _("It's not possible to prorate daily periods."))
+        qty = int(math.ceil(qty)) * self.membership_interval_qty
+        if self.membership_interval_unit == 'weeks':
+            weekday = date.weekday()
+            date_from = date - datetime.timedelta(weekday)
```

Comment:
Perhaps can do it more 'Odooish' using 'date_utils'

Reply:
Thanks @Tardo I refactored it using date_utils as you suggested ;)